pkg/trdm/header.go

```diff
@@ -149,11 +198,24 @@ func GenerateSignedHeader(certificate *x509.Certificate, privateKey *rsa.Private
 		return nil, err
 	}
 	msgHashSum := msgHash.Sum(nil)
-
 	signedHash, err := privateKey.Sign(rand.Reader, msgHashSum, crypto.SHA512)
 	if err != nil {
 		return nil, err
 	}
+	// publicKey, ok := certificate.PublicKey.(*rsa.PublicKey)
```

Comment:
We don't, but I'm currently sitting on it until we get a successful TRDM hit from the golang envelope - it could be based on how we're using the keys to sign. For example, using privateKey.Sign() vs rsa.EncryptPKCS1v15(). I'll make sure it's all cleaned up from the parent branch when that PR goes up - `B-17248-Soap-Security-Header`